### PR TITLE
Element-R: Basic implementation of SAS verification

### PR DESCRIPTION
**Pull request by richvdh**
Fixes https://github.com/vector-im/element-web/issues/25316

~~Based on https://github.com/matrix-org/matrix-js-sdk/pull/3495~~

Notes: Support for interactive device verification in Element-R.

---
Here's what your changelog entry will look like:

## ✨ Features
 * Support for interactive device verification in Element-R. ([\#3490](https://github.com/matrix-org/matrix-js-sdk/pull/3490)). Fixes vector-im/element-web#25316.

### CODE REVIEW

File: src/rust-crypto/rust-crypto.ts [hunk 1]
```diff
@@ -517,7 +525,18 @@ export class RustCrypto implements CryptoBackend {
...
+        const user: RustSdkCryptoJs.OwnUserIdentity = await this.olmMachine.getIdentity(
+            new RustSdkCryptoJs.UserId(this.userId),
+        );
+        const [request, outgoingRequest]: [RustSdkCryptoJs.VerificationRequest, RustSdkCryptoJs.ToDeviceRequest] =
+            await user.requestVerification(
+                this.supportedVerificationMethods?.map(verificationMethodIdentifierToMethod),
+            );
+        await this.outgoingRequestProcessor.makeOutgoingRequest(outgoingRequest);
+        return new RustVerificationRequest(request, this.outgoingRequestProcessor);
+        */
```
Comment (SimonBrandner) [inline src/rust-crypto/rust-crypto.ts:539]: I am not sure how we feel about leaving code like this in our codebase. Might be enough it's in the GIT history now?

**Review by SimonBrandner — APPROVED** [review]
Looks sane

File: src/rust-crypto/rust-crypto.ts (hunk 1 above)
Comment (richvdh) [reply src/rust-crypto/rust-crypto.ts:539]: yes, fair. I'll rip it out for now.
```suggestion
```

File: src/rust-crypto/rust-crypto.ts [hunk 2]
```diff
@@ -571,7 +579,18 @@ export class RustCrypto implements CryptoBackend {
...
```
Comment (SimonBrandner) [inline src/rust-crypto/rust-crypto.ts:593]: I think you somehow failed to rip this out - perhaps a merge fail?

**Review by SimonBrandner — APPROVED** [review]

File: src/rust-crypto/rust-crypto.ts (hunk 2 above)
Comment (richvdh) [reply src/rust-crypto/rust-crypto.ts:593]: argh. yes. thanks.

**Review by andybalaam — APPROVED** [review]

**Pull request merged**